packages/flutter/lib/src/rendering/object.dart

```diff
@@ -3916,3 +3916,20 @@ class DiagnosticsDebugCreator extends DiagnosticsProperty<Object> {
         level: DiagnosticLevel.hidden,
       );
 }
+
+/// A mixin for [RenderObject] subclasses that can configure its clip behavior.
+mixin ClipBehaviorMixin on RenderObject {
+  /// {@macro flutter.widgets.Clip}
+  ///
+  /// Defaults to [Clip.none], and must not be null.
+  Clip get clipBehavior => _clipBehavior;
+  Clip _clipBehavior = Clip.none;
+  set clipBehavior(Clip value) {
+    assert(value != null);
+    if (value != _clipBehavior) {
+      _clipBehavior = value;
+      markNeedsPaint();
+      markNeedsSemanticsUpdate();
+    }
+  }
+}
```

Comment:
this doesn't belong in `object.dart`. It's way too specific.

Is there any reason we wouldn't just duplicate this code everywhere rather than using mixins? Mixins aren't free (they add more classes to our class hierarchy, etc). I don't know how well they interact with tree shaking either. We don't have any trouble duplicating other fields. Why is this one special?

Reply:
That will also fix the documentation trouble I mentioned in the other comment.